packages/compass-components/src/hooks/use-confirmation.tsx

```diff
@@ -0,0 +1,114 @@
+import React, { useContext, useEffect, useState } from 'react';
+
+import {
+  default as ConfirmationModal,
+  Variant as ModalVariant,
+} from '@leafygreen-ui/confirmation-modal';
+
+export { ModalVariant };
+
+type ConfirmationProperties = {
+  title: string;
+  description: string;
+  buttonText?: string;
+  variant?: ModalVariant;
+};
+
+interface ConfirmationModalContextData {
+  showConfirmation: (props: ConfirmationProperties) => Promise<boolean>;
+}
+
+const ConfirmationModalContext =
+  React.createContext<ConfirmationModalContextData>(
+    {} as ConfirmationModalContextData
+  );
+
+class ShowConfirmationEvent extends Event {
+  constructor(
+    public props: ConfirmationProperties,
+    public callback: (value: boolean) => void = () => {}
+  ) {
+    super('show');
+  }
+}
+class GlobalConfirmation extends EventTarget {
+  showConfirmation(props: ConfirmationProperties) {
+    return new Promise<boolean>((resolve) => {
+      const event = new ShowConfirmationEvent(props, resolve);
+      this.dispatchEvent(event);
+    });
+  }
+}
+const globalConfirmation = new GlobalConfirmation();
+export const showConfirmation =
+  globalConfirmation.showConfirmation.bind(globalConfirmation);
+
+export const ConfirmationModalArea: React.FC = ({ children }) => {
+  const [isOpen, setIsOpen] = useState(false);
+  const [data, setData] = useState({} as ConfirmationProperties);
+  const [callback, setCallback] = useState<(value: boolean) => void>(() => {});
+
+  useEffect(() => {
+    return () => {
+      setIsOpen(false);
+      setData({} as ConfirmationProperties);
+      setCallback(() => {});
+    };
+  }, []);
+
+  // Event listener to use confirmation modal outside of react
+  useEffect(() => {
+    const listener = (event: ShowConfirmationEvent) => {
+      showConfirmation(event.props).then(event.callback);
+    };
+
+    globalConfirmation.addEventListener('show', listener);
+
+    return () => {
+      globalConfirmation.removeEventListener('show', listener);
+    };
+  }, []);
+
+  const showConfirmation = (props: ConfirmationProperties) => {
```

Comment:
If you are passing something down React tree with context, you need to be extra careful about making the value as stable as possible as changed `value` property on a Provider will cause re-renders in all components that access the context. In this case every time this provider re-renders, it will cause cascading re-renders for places that will be using this context